### ISSUE TITLE
Comparing identical `Data` objects with `==` returns `False`

### ISSUE THREAD

**Issue by callumm-graphcore** [pyg-team/pytorch_geometric#5991]
### 🐛 Describe the bug

If I have two graphs with identical contents, and compare them with `==`, it returns `False`.

For example, the following code:
```python
import torch
from torch_geometric.data import Data

x = torch.rand(30, 50)
edge_index = torch.randint(size=[2, 70], high=30)

d1 = Data(x=x, edge_index=edge_index)
d2 = Data(x=x, edge_index=edge_index)

print((d1.x == d2.x).all())
print((d1.edge_index == d2.edge_index).all())

print(d1 == d2)
```
prints:
```
True
True
False
```

It appears that this happens because `Data` inherits its `__eq__` method from `FeatureStore`, which implements this method by comparing the `id()` of each object, so two otherwise identical objects will not be considered equal.

Please let me know if there is any more information I can provide. Thank you for your time.

### Environment

* PyG version: 2.1.0
* PyTorch version: 1.10.0+cpu
* OS: Ubuntu 20.04
* Python version: 3.8.10
* CUDA/cuDNN version: N/A
* How you installed PyTorch and PyG (`conda`, `pip`, source): PyTorch from `pip`, PyG from master (I think?)
* Any other relevant information (*e.g.*, version of `torch-scatter`): I don't think the versions of `torch-scatter` etc are relevant here

**Comment by rusty1s** [comment]
Yes, this is a good point. We should probably fix the `__eq__` implementation here. Do you have interest in contributing this?

**Comment by callumm-graphcore** [comment]
I'd be interested, but I'm pretty busy with work at the moment. Is there a particular time by which you'd like to see this fixed?

**Comment by rusty1s** [comment]
No, this is not urgent. Just let me know :)
